pkg/sbproxy/reconcile/reconcile_brokers.go

```diff
@@ -208,8 +209,12 @@ func mapByURLSmIDSuffix(predicate filterBrokerPredicate) func(platform.ServiceBr
 	}
 }
 
-func mapByWholeURL(broker platform.ServiceBroker) (string, bool) {
-	return broker.BrokerURL, true
+func mapByNameAndWholeURL(broker platform.ServiceBroker) (string, bool) {
+	return getBrokerNameURLCombination(&broker), true
+}
+
+func getBrokerNameURLCombination(broker *platform.ServiceBroker) string {
+	return broker.Name + "-" + broker.BrokerURL
```

Comment:
It would be safer to keep these separate in a struct and use the struct as a key in the map, e.g.
```go
type brokerKey struct {
  name string
  url string
}
```